extensions/sql/contract-definition-store-sql/src/main/java/org/eclipse/dataspaceconnector/sql/contractdefinition/store/SqlContractDefinitionStore.java

```diff
@@ -69,7 +69,15 @@ ContractDefinition mapResultSet(ResultSet resultSet) throws Exception {
 
     @Override
     public @NotNull Collection<ContractDefinition> findAll() {
-        return findAll(QuerySpec.none()).collect(Collectors.toList());
+        return transactionContext.execute(() -> {
```

Comment:
I would make this more explicit as it was done in the `SqlAssetIndex` at line 132, passing to the other `findAll` method a `QuerySpec` instance with an insanely high `limit` value.
In this way the thing will be explicit (as I would expect this to change in the future), there will be no duplication and no SQL syntax will overflow in the store implementation

Reply:
There isn't really any duplication, there is just two (slightly) different SQL statements. In reality those two approaches would have the same effect, and once we have true pagination in IDS, we can remove the parameterless `findAll()` method completely.

What do you mean by "SQL syntax overflow"?

Reply:
having a string with `SELECT something FROM somewhere` in the `Store`/`Index`, we never did that anywhere else, all the SQL stuff is in the `Statements` class

Reply:
ah got it. How about we remove `findAll()` altogether then, and provide a static method `QuerySpec.max()` that creates a QuerySpec with `limit = Integer.MAX_VALUE`?

Reply:
[this commit](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1539/commits/a73fa531f3b37a7175912eb5d1f1898514a3dc7b) removes the parameterless `findAll()`. WDYT?